src/main/java/org/jpeek/graph/Disjoint.java

```diff
@@ -48,6 +50,26 @@ public Disjoint(final Graph graph) {
 
     @Override
     public List<Set<Node>> value() throws Exception {
-        throw new UnsupportedOperationException(this.graph.toString());
+        final Set<Node> unvisited = new HashSet<>(this.graph.nodes());
+        final List<Set<Node>> result = new ArrayList<>(unvisited.size());
+        while (!unvisited.isEmpty()) {
+            final Node node = unvisited.iterator().next();
+            final Set<Node> adding = new HashSet<>();
```

Comment:
@HDouss Honestly `adding `variable is very confusing here, hard to catch what it is for.
Could you please rename so that others can easily understand?